misk/src/main/kotlin/misk/web/interceptors/ConcurrencyLimitsInterceptor.kt

```diff
@@ -82,17 +110,34 @@ internal class ConcurrencyLimitsInterceptor internal constructor(
     private val clock: Clock,
     private val limiterFactories: List<ConcurrencyLimiterFactory>,
   ) : NetworkInterceptor.Factory {
+    /**
+     * Note that this cache is application-global. Multiple actions that use the same Quota-Path
+     * will be treated as a homogenous group for concurrency limiting.
+     */
+    private val quotaPathToLimiter: Cache<String, Limiter<String>> = CacheBuilder.newBuilder()
+      .build()
+
     override fun create(action: Action): NetworkInterceptor? {
       if (action.function.findAnnotation<AvailableWhenDegraded>() != null) return null
+      return ConcurrencyLimitsInterceptor(
+        factory = this,
+        action = action,
+        defaultLimiter = createLimiterForAction(action),
+        clock = clock
+      )
+    }
 
-      val limiter = limiterFactories.asSequence()
+    private fun createLimiterForAction(action: Action): Limiter<String> {
+      return limiterFactories.asSequence()
         .mapNotNull { it.create(action) }
         .firstOrNull()
-        ?: SimpleLimiter.Builder()
+        ?: Builder()
```

Comment:
supernit: I prefer keeping `SimpleLimiter.Builder()` bc it is more explicit and readable.

Reply:
totally agree. IntelliJ did this behind my back